src/nvim/tui/tui.c

```diff
@@ -641,12 +645,21 @@ static void update_size(UI *ui)
 {
   TUIData *data = ui->data;
   int width = 0, height = 0;
-  // 1 - try from a system call(ioctl/TIOCGWINSZ on unix)
+
+  // 1 - if starting and the 'columns' and 'lines' options are not
+  // the defaults, use their values
+  if (starting && (Columns != 80 || Rows != 24)) {
```

Comment:
Let me see what I can do. It would be best if we defined sth that would get used here and in `option.c`.